test/differentials/composite.jl

```diff
@@ -18,6 +18,15 @@ struct StructWithInvariant
     StructWithInvariant(x) = new(x, 2x)
 end
 
+function _unpack2tuple(comp)
+    a, b = comp
+    return (a, b)
+end
+
+function _unpacknamedtuple(comp)
+    x, y = comp.x, comp.y
+    return (x, y)
+end
```

Comment:
Do you think we should move these into the place they are used?
Or even make then anon functions?